tensorflow_addons/optimizers/average_wrapper.py

```diff
@@ -45,6 +47,12 @@ def __init__(
         self._optimizer = optimizer
         self._sequential_update = sequential_update
 
+        warnings.warn(
```

Comment:
Here all the users will see the warning, even if they've never used it. To target exactly the users who haved changed the parameter, set the default value to None. Then do:

```python
if sequential_update is not None:
    warnings.warn(...)
```

Reply:
Unfortunately, the default value for sequential_update was set to `True` when it was added, and that has remained.

So testing for not None is not effective. Since the type of sequential_update is strictly Bool.

Reply:
We can relax the bool requirement by typing it as Optional[bool], then we can set it to None by default and the warning will trigger only if the user had been setting the value manually. Do you think it makes sense?

Reply:
Yeah, made it `Optional[bool]`. I removed `sequential_update` from the optimizer's config as well. Loading an older model with `sequential_update` will show the `DeperecationWarning`.